zerver/lib/actions.py

```diff
@@ -817,6 +817,44 @@ def do_send_messages(messages):
     # intermingle sending zephyr messages with other messages.
     return already_sent_ids + [message['message'].id for message in messages]
 
+def do_send_typing_notification(notification, stopped):
+    # type: (Dict[str, Any], bool) -> None
+    if stopped is True:
+        event_type = 'stopped_typing_notification'
+    else:
+        event_type = 'typing_notification'
+    event = dict(
+            type            = event_type,
+            sender          = notification['sender'],
+            recipient       = notification['recipient'])
```

Comment:
I'd probably use `type = 'typing'` and `op='start'` / `op='stop'` to better match the style of our other events.